cpp/ql/lib/semmle/code/cpp/commons/StringConcatenation.qll

```diff
@@ -0,0 +1,98 @@
+/**
+ * A library for detecting general string concatenations.
+ */
+
+import cpp
+import semmle.code.cpp.models.implementations.Strcat
+import semmle.code.cpp.models.interfaces.FormattingFunction
+import semmle.code.cpp.dataflow.new.DataFlow
```

Comment:
I don't think we should have people rely on dataflow implicitly being imported when they import this library. Ideally, the other imports should also be private, but if you're using a library for string concatenation I think it's fair that you also implicitly import those other libraries.

```suggestion
private import semmle.code.cpp.dataflow.new.DataFlow
```